### ISSUE TITLE
X-Registry-Auth Header for Service

### ISSUE THREAD

**Issue by stefanthorpe** [docker-php/docker-php#229]
According to the docs, I should send X-Registry-Auth Headers so private images can be pulled across a swarm.
I've looked through the code and can't see the support for this. I can see it in the Image manager but not ServiceManager.
Am I reading the code correctly or am I overlooking something?

**Comment by stefanthorpe** [comment]
Have done some more testing and X-Registry-Auth isn't currently working in serviceManager. Package returns the following exception.
```
UndefinedOptionsException in OptionsResolver.php line 685:
The option "X-Registry-Auth" does not exist. Defined options are: "".
```

**Comment by stefanthorpe** [comment]
#231 fixes the issue

**Comment by joelwurtz** [comment]
Parameters is definitely available in 2.0 see : https://github.com/docker-php/docker-php-api/blob/4.1.36/src/Endpoint/ServiceUpdate.php#L29
